modules/stanford_earth_events/stanford_earth_events.module

```diff
@@ -72,6 +76,30 @@ function stanford_earth_events_migrate_prepare_row(Row $row, MigrateSourceInterf
     if ($event->isValid()) {
       $entityid = $event->entityId();
       if (!empty($entityid)) {
+        $eventNode = Node::load($entityid);
+
+        // If the event node has a media entity image but no image_url in the
+        // event itself, the image will get removed unless we set the source
+        // image_url (field_s_event_image) to the url of the current image.
+        if (empty($row->getSourceProperty('field_s_event_image')) &&
+          !empty($eventNode->get('field_s_event_media')->getValue())) {
+          $mids = $eventNode->get('field_s_event_media')->getValue();
```

Comment:
since the field only allows 1 value, you could do `$eventNode->get('field_s_event_media')->get(0)`. This would allow you to cut out the `foreach` in the following line.